kinto/views/__init__.py

```diff
@@ -29,4 +30,10 @@ def object_exists_or_404(request, collection_id, object_id, parent_id=''):
                            object_id=object_id)
     except exceptions.RecordNotFoundError:
         # XXX: We gave up putting details about parent id here (See #53).
-        raise httpexceptions.HTTPNotFound()
+        detail_dict = {
+                "id": collection_id,
+                "resource_name": "collection"
+            }
```

Comment:
Here `"id"` should have value of `object_id` and `resource_name` value of `collection_id` (ref #710)